appinventor/blocklyeditor/src/backpack.js

```diff
@@ -310,13 +314,15 @@ Blockly.Backpack.prototype.checkValidBlockTypes = function(block, arr) {
  *
  */
 Blockly.Backpack.prototype.addAllToBackpack = function() {
-  var allBlocks = Blockly.mainWorkspace.getAllBlocks();
   var topBlocks = Blockly.mainWorkspace.getTopBlocks(false);
   var p = this;
   this.getContents(function(contents) {
+    var saveAsync = p._NoAsync;
```

Comment:
While I doubt this could happen, an exception thrown by one of the Blockly.Xml methods called in `addToBackpack` could result in an exception, in which case `_NoAsync` will not be reset at the end of this function. Think about whether it is worth adding a `try-finally` block.